src/dbwrapper.h

```diff
@@ -254,6 +205,11 @@ class CDBWrapper
     //! whether or not the database resides in memory
     bool m_is_memory;
 
+    bool ReadImpl(std::function<void(DataStream&)> write_key_to_stream, std::function<void(CDataStream&)> read_value_from_stream) const;
+    bool ExistsImpl(const Span<const std::byte>& key) const;
+    size_t EstimateSizeImpl(const Span<const std::byte>& key1, const Span<const std::byte>& key2) const;
+    auto& DBContext() const { return *Assert(m_db_context); }
```

Comment:
nit: I think this needs a `LIFETIMEBOUND`.

Also: suggested alternative name `DBContextRef()` to highlight that the return value can mutate our `CDBWrapper` state? No strong view on it though, happy with the current one too, just a suggestion.